tel2puml/otel_to_pv/data_holders/sql_data_holder/sql_dataholder.py

```diff
@@ -230,12 +226,77 @@ def find_unique_graphs(self) -> dict[str, set[str]]:
         :return: A dictionary mapping job names to a set of unique job_ids
         :rtype: `dict`[`str`, `set`[`str`]]
         """
-        return find_unique_graphs(
-            self.time_buffer,
-            self.batch_size,
-            self
+        return find_unique_graphs(self.time_buffer, self.batch_size, self)
+
+    def stream_job_name_batches(
+        self,
+        session: Session,
+        job_name_to_job_ids_map: dict[str, set[str]] | None = None,
+        filter_job_names: set[str] | None = None,
+    ) -> Generator[OTelEvent, None, None]:
+        """
+        Stream OTelEvents from the database.
+
+        :param session: SQLAlchemy Session object.
+        :type session: :class: `sqlalchemy.orm.Session`
+        :param job_name_to_job_ids_map: Optional mapping of job names to
+        job IDs to filter.
+        :type job_name_to_job_ids_map: `dict`[`str`, `set`[`str`]] | `None`
+        :param filter_job_names: Optional set of job names to filter.
```

Comment:
In future can you put what it defaults as thats how you write sphinx type docs i.e. :param filter_job_names: Optional set of job names to filter, defaults to None
and below :type filter_job_names: `set`[`str`] | `None`, optional
can you change this throughout